mrbgems/mruby-io/src/io.c

```diff
@@ -573,15 +575,8 @@ mrb_io_initialize_copy(mrb_state *mrb, mrb_value copy)
   buf = mrb_iv_get(mrb, orig, mrb_intern_cstr(mrb, "@buf"));
   mrb_iv_set(mrb, copy, mrb_intern_cstr(mrb, "@buf"), buf);
 
-  fptr_copy->fd = mrb_dup(mrb, fptr_orig->fd, &failed);
-  if (failed) {
-    mrb_sys_fail(mrb, 0);
-  }
-  fptr_copy->fd2 = mrb_dup(mrb, fptr_orig->fd2, &failed);
-  if (failed) {
-    close(fptr_copy->fd);
-    mrb_sys_fail(mrb, 0);
-  }
+  fptr_copy->fd = mrb_cloexec_dup(mrb, fptr_orig->fd);
+  fptr_copy->fd2 = mrb_cloexec_dup(mrb, fptr_orig->fd2);
```

Comment:
The old code prevents fd leak.

Reply:
Isn't GC enough to close fd?

Reply:
I think it's good to prevent resource leaks as early as possible.

Reply:
OK, I fixed it.